### ISSUE TITLE
Colors from child scripts not being used in logged output

### ISSUE THREAD

**Issue by kentcdodds** [kimmobrunfeldt/concurrently#86]
Thanks for this tool! I think it's great! I'm not getting colorized output when running concurrently on a script that is outputing stuff with chalk. Here's a repo that reproduces the issue: https://github.com/kentcdodds/chalk-concurrently-bug

<img width="647" alt="output" src="https://cloud.githubusercontent.com/assets/1500684/22663109/908c2c42-ec68-11e6-8c9c-75b165f2f5db.png">

It's pretty simple. `color-red.js` is:

```javascript
#!/usr/bin/env node
const chalk = require('chalk')

console.log(chalk.red(process.argv.slice(2).join(' ')))
```

Any ideas? This is sort of a blocker for [`p-s`](https://github.com/kentcdodds/p-s) dropping the `--parallel` option and recommending people just use `concurrently`. ([more info](https://github.com/kentcdodds/p-s/pull/94)).

**Comment by gustavohenke** [comment]
Hi @kentcdodds! Thanks for the report.
After some research, I think that this is a standard behavior of Node itself.

See:
- http://stackoverflow.com/questions/18825493/retaining-output-colors-when-shelling-out-to-node
- http://stackoverflow.com/questions/7725809/preserve-color-when-executing-child-process-spawn

You can, however, use the `--raw` flag. Under the hoods, the behavior of `--raw` is to simply use `stdio: "inherit"`, when spawning the child process.
I was able to always retain the colors when using it, even though the output is less cooler :)

**Comment by kentcdodds** [comment]
Ah! I found [this](http://stackoverflow.com/a/31621617/971592) which explains what's going on quite nicely.

I think that a solution could be to simply add `FORCE_COLOR = 1` to the `env` that you spawn the script with. Right around [here](https://github.com/kimmobrunfeldt/concurrently/blob/5a0d525af57ea2513eee61321af05317ba909411/src/main.js#L184) we could probably change that to:

```javascript
var spawnOpts = config.raw
  ? {stdio: 'inherit'}
  : {
      env: Object.assign({FORCE_COLOR: 1}, process.env)
    };
```

I'd be happy to make a PR to do this!

**Comment by okonet** [comment]
You can check if it's TTY and only force colors in this case. This solved CI and similar cases:

https://github.com/okonet/lint-staged/blob/master/src/index.js#L18

**Comment by kentcdodds** [comment]
We could just use https://www.npmjs.com/package/supports-color :) That's the same thing that chalk uses.

```javascript
var spawnOpts = config.raw
  ? {stdio: 'inherit'}
  : {
      env: Object.assign({FORCE_COLOR: Boolean(supportsColor)}, process.env)
    };
```

**Comment by gustavohenke** [comment]
Exactly what I thought 👍 

If you have the time to send such PR, please go ahead.
Otherwise I'll do it later and push a new version.

**Comment by kentcdodds** [comment]
Writing tests for this is pretty difficult and would require a bit of a rewrite of how tests work. So I'm not going to add tests to my changes.

**Comment by gustavohenke** [comment]
Published [v3.2.0](https://github.com/kimmobrunfeldt/concurrently/releases/tag/3.2.0)!

**Comment by sinedied** [comment]
Using concurrently `8.8.2` I don't have any colors from my child scripts, unless I use `--raw`. Is this a regression?
